src/integration/Sites.spec.ts

```diff
@@ -0,0 +1,205 @@
+import express from "express"
+import mockAxios from "jest-mock-axios"
+import request from "supertest"
+
+import { IsomerAdmin, Repo, Site, SiteMember, User } from "@database/models"
+import { generateRouter } from "@fixtures/app"
+import UserSessionData from "@root/classes/UserSessionData"
+import { mockEmail, mockIsomerUserId } from "@root/fixtures/sessionData"
+import { SitesRouter as _SitesRouter } from "@root/routes/v2/authenticated/sites"
+import { genericGitHubAxiosInstance } from "@root/services/api/AxiosInstance"
+import { GitHubService } from "@root/services/db/GitHubService"
+import { ConfigYmlService } from "@root/services/fileServices/YmlFileServices/ConfigYmlService"
+import IsomerAdminsService from "@root/services/identity/IsomerAdminsService"
+import { SitesService } from "@root/services/utilServices/SitesService"
+import { getUsersService } from "@services/identity"
+import { sequelize } from "@tests/database"
+
+const mockSite = "mockSite"
+const mockSiteId = "1"
+const mockBlockedSite = "adminOnly"
+const mockUpdatedAt = "now"
+const mockPermissions = { push: true }
+const mockPrivate = true
+
+const gitHubService = new GitHubService({ axiosInstance: mockAxios })
+const configYmlService = new ConfigYmlService({ gitHubService })
+const usersService = getUsersService(sequelize)
+const isomerAdminsService = new IsomerAdminsService({ repository: IsomerAdmin })
+const sitesService = new SitesService({
+  gitHubService,
+  configYmlService,
+  usersService,
+  isomerAdminsService,
+})
+
+const SitesRouter = new _SitesRouter({ sitesService })
+const sitesSubrouter = SitesRouter.getRouter()
+
+// Set up express with defaults and use the router under test
+const subrouter = express()
+// As we set certain properties on res.locals when the user signs in using github
+// In order to do integration testing, we must expose a middleware
+// that allows us to set this properties also
+subrouter.use((req, res, next) => {
+  const userSessionData = new UserSessionData({
+    isomerUserId: mockIsomerUserId,
+    email: mockEmail,
+  })
+  res.locals.userSessionData = userSessionData
+  next()
+})
+subrouter.use(sitesSubrouter)
+const app = generateRouter(subrouter)
+
+describe("Sites Router", () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+    mockAxios.reset()
+  })
+
+  describe("/", () => {
```

Comment:
could we have a few more test cases so that these invariants are clear

1. for github users, their access token only gives them access to their site
2. sites w/o `siteMembers` are seen by admins (i think this is so for `blockedSite`? if so, maybe just rename it to make it clearer)
3. public sites can be seen by everyone (<-- not q sure what should be the behaviour here, correct me if i'm wrong)

Reply:
1. I left this one out of the integration tests because it's covered by the unit tests - the Github API call is the one which restricts the sites accessible to the user, so I didn't think there was much benefit in including it here - we'd just be mocking the Github results and seeing that we receive the same mocked results.
2. Yup that was the purpose of `blockedSite` - I've renamed it to `mockAdminSite` if that's clearer? in 1a88af5, lmk if there's a better name for it
3. Hmm I think there's no such thing as public sites really - sites are "public" in that anyone can view them, but access to sites are still restricted based on whether someone has _write_ permissions to the site, which is what the getSites method is returning

Reply:
yeap sure, 1/2 seems reasonable! so just double checking for 3 - when we call `getSites`, this is restricted based on _write_ access and not _read_ access? this is because all of our sites are public (on github) and when we say write access, in this context what it means is **users who are members of that github organization that own the site**? think the semantics of this needs to be nailed down HAHA

Reply:
Yup! To clarify - the github repos are all public, and anyone can access them via github directly. On our CMS, only sites that a user is allowed to write to are viewable - for github login users, that's via checking push permissions (`repoData.permissions.push === true`), and for email login, via checking the `site_members` table

Reply:
ok, thanks for clarifying!